### PR TITLE
AG-12836 - Open 'text' tab in options dialog when trend line label te…

### PR DESCRIPTION
**Pull request by manapeirov**
PR for https://ag-grid.atlassian.net/browse/AG-12836

### CODE REVIEW

File: packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts [hunk 1]
```diff
@@ -333,8 +333,14 @@ export class AnnotationsStateMachine extends StateMachine<States, AnnotationType
 
                 dblclick: {
                     guard: guardActiveHasLineText,
-                    action: () => {
-                        ctx.showAnnotationSettings(this.active!, undefined);
+                    action: ({ offset }) => {
+                        const nodeAtCoords = ctx.getNodeAtCoords(offset, this.active!);
+
+                        if (!(nodeAtCoords === 'line' || nodeAtCoords === 'text')) {
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts]:
```suggestion
                        if (nodeAtCoords !== 'line' && nodeAtCoords !== 'text') {
```

Generally, it's a bit more readable like this, as the `(!(` can be missed when scanning the code.

Comment (lsjroberts) [reply packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts]: Also, I'd guess we will want to open on the "line" tab when the user double clicks a handle? So could do something like `const node = nodeAtCoords === 'text' ? 'text' : 'line';`

**Review by lsjroberts — APPROVED** [review]
LGTM, with one small comment.

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**